### DISCUSSION TITLE
Workaround for Compile Error "esp32-arduino requires CONFIG_FREERTOS_HZ=1000 (currently 100)"

### DISCUSSION THREAD

**Discussion by adrianeccles86** [espressif/arduino-esp32#8375] [Q&A]
Hi,

I am using esp32-arduino with Platform.io and have some code that uses android framework components and espressive framework components.

I am struggling to to get the code to compile due to error "esp32-arduino requires CONFIG_FREERTOS_HZ=1000 (currently 100)". 
I have done some digging on the internet but have not yet found a solution/work around.

Can anyone help.

Thanks,
Adrian

**Comment by me-no-dev** [comment]
add `CONFIG_FREERTOS_HZ=1000` to a file called `sdkconfig.defaults` in your project root

**Reply by cemox** [reply]
this does not work for me. when idf.py menuconfig runs, it overrides sdkconfig content. (in my situation) (IDF 5.1.12). So menuconfig cannot be started, stuck in a loop.

**Comment by SylvianHemus** [comment]
FWIW, I had to add it to sdkconfig, not sdkconfig.defaults

**Reply by sohamc1** [reply]
This indeed solves the problem.

**Comment by cemox** [comment]
Temporarily changing the value from 100 to 1000 in components/cmakelists.txt solves the problem and menuconfig runs wtihout error.

After that change the parameter in menuconfig and save the sdkconfig file. 

Lastly, revert cmakelists.txt file to its original state (i.e. undo the change you make)

```
if(NOT CONFIG_FREERTOS_HZ EQUAL 1000 AND NOT "$ENV{ARDUINO_SKIP_TICK_CHECK}")
    # See delay() in cores/esp32/esp32-hal-misc.c.
    message(FATAL_ERROR "esp32-arduino requires CONFIG_FREERTOS_HZ=1000 "
                        "(currently ${CONFIG_FREERTOS_HZ})")
endif()  
```

original should be like:

```
if(NOT CONFIG_FREERTOS_HZ EQUAL 100 AND NOT "$ENV{ARDUINO_SKIP_TICK_CHECK}")
    # See delay() in cores/esp32/esp32-hal-misc.c.
    message(FATAL_ERROR "esp32-arduino requires CONFIG_FREERTOS_HZ=1000 "
                        "(currently ${CONFIG_FREERTOS_HZ})")
endif()
```
pay attention to bold part " ... NOT CONFIG_FREERTOS_HZ EQUAL **100** ..."
